subsys/bluetooth/mesh/rpl.c

```diff
@@ -210,22 +212,23 @@ void bt_mesh_rpl_reset(void)
 
 		if (rpl->src) {
 			if (rpl->old_iv) {
-				if (IS_ENABLED(CONFIG_BT_SETTINGS)) {
+				if (IS_ENABLED(CONFIG_BT_MESH_RPL_STORE)) {
 					clear_rpl(rpl);
 				} else {
 					(void)memset(rpl, 0, sizeof(*rpl));
 				}
 			} else {
 				rpl->old_iv = true;
 
-				if (IS_ENABLED(CONFIG_BT_SETTINGS)) {
+				if (IS_ENABLED(CONFIG_BT_MESH_RPL_STORE)) {
 					schedule_rpl_store(rpl, true);
 				}
 			}
 		}
 	}
 }
 
+#ifdef CONFIG_BT_MESH_RPL_STORE
```

Comment:
If you convert this macro to generic one: https://github.com/zephyrproject-rtos/zephyr/blob/c050d1f2ecb60a366ab6e2aa9098d9cc23ee536d/subsys/bluetooth/mesh/settings.h#L33-L37 and use it when `CONFIG_BT_MESH_RPL_STORE` is not enabled, then there is no need to wrap the code with `#ifdef ... #endif``.

Something like this:
https://github.com/PavelVPV/zephyr/commit/2735e992a6f22a9ff1cfe6983a334cf748612008